build-all.sh

```diff
@@ -65,28 +72,42 @@ if [ -e "$BUILDSTATUS_FILE" ]; then
 	echo "Continuing build-all from: $BUILDSTATUS_FILE"
 fi
 
-exec >	>(tee -a "$BUILDALL_DIR"/ALL.out)
-exec 2> >(tee -a "$BUILDALL_DIR"/ALL.err >&2)
-trap 'echo ERROR: See $BUILDALL_DIR/${PKG}.err' ERR
+exec &>	>(tee -a "$BUILDALL_DIR"/ALL.out)
+trap 'echo ERROR: See $BUILDALL_DIR/${PKG}.out' ERR
 
 while read -r PKG PKG_DIR; do
 	# Check build status (grepping is a bit crude, but it works)
-	if [ -e "$BUILDSTATUS_FILE" ] && grep "^$PKG\$" "$BUILDSTATUS_FILE" >/dev/null; then
+	if [ -e "$BUILDSTATUS_FILE" ] && grep -q "^$PKG\$" "$BUILDSTATUS_FILE"; then
 		echo "Skipping $PKG"
 		continue
 	fi
 
+	# Start building
+	if [ -n "${TERMUX_DEBUG_BUILD}" ]; then
+		echo "\"$BUILDSCRIPT\" -a \"$TERMUX_ARCH\" $TERMUX_DEBUG_BUILD --format \"$TERMUX_FORMAT\" --library $(package__is_package_name_have_glibc_prefix "$PKG" && echo "glibc" || echo "bionic") ${TERMUX_OUTPUT_DIR+-o $TERMUX_OUTPUT_DIR} $TERMUX_INSTALL_DEPS \"$PKG_DIR\""
+	fi
 	echo -n "Building $PKG... "
 	BUILD_START=$(date "+%s")
-	bash -x "$BUILDSCRIPT" -a "$TERMUX_ARCH" $TERMUX_DEBUG_BUILD \
+	"$BUILDSCRIPT" -a "$TERMUX_ARCH" $TERMUX_DEBUG_BUILD --format "$TERMUX_FORMAT" \
+		--library $(package__is_package_name_have_glibc_prefix "$PKG" && echo "glibc" || echo "bionic") \
 		${TERMUX_OUTPUT_DIR+-o $TERMUX_OUTPUT_DIR} $TERMUX_INSTALL_DEPS "$PKG_DIR" \
-		> "$BUILDALL_DIR"/"${PKG}".out 2> "$BUILDALL_DIR"/"${PKG}".err
+		&> "$BUILDALL_DIR"/"${PKG}".out
 	BUILD_END=$(date "+%s")
 	BUILD_SECONDS=$(( BUILD_END - BUILD_START ))
-	echo "done in $BUILD_SECONDS"
+	echo "done in $BUILD_SECONDS sec"
 
 	# Update build status
 	echo "$PKG" >> "$BUILDSTATUS_FILE"
+
+	# Check which packages were also compiled
+	if [ -z "$TERMUX_INSTALL_DEPS" ]; then
+		for build_pkg in $(find ~/.termux-build/ -mindepth 1 -maxdepth 1 -type d | grep -v -e '^_cache' -e '^_buildall'); do
```

Comment:
This is the way to use `find` with `null` terminator `-print0`. If piping to `grep`, then pass `-z` option to it.

https://github.com/termux/termux-packages/blob/f97955f1ec0a3f1cd2a2ce32903672502914704e/scripts/build/termux_step_massage.sh#L110-L115

Reply:
I'd be fine with that as well.
But just looping over the glob is gonna be much simpler here.

Reply:
Will have to do directory check and `grep` inside loop then. Glob normally is fine too, but it does not match hidden files starting with `.` by default, hence `find` is mostly the way to go.

Reply:
`ls -1` for all its other flaws also does not find dotfiles.

If you wanted it with a glob you can use `{\.,}*`

Reply:
> The issue here is that any of the regular word splitting concerns about spaces, tabs and newlines.

This would make sense if `~/.termux-build/` could create files/folders with names that are difficult to parse for posix. But usually there are folders there (with package names and system folders for the build script) that do not break parsing via `ls/find`.

> Please just loop over the directory directly - using globbing - and do any additional checks you feel you need on "$build_pkg" which will at that point contain the sanitized file names.

I doubt this solution because here you will have to put an additional check via `grep` inside the loop, which in turn will increase the "scanning" time.